rig/place_and_route/utils.py

```diff
@@ -360,6 +364,54 @@ def build_routing_tables(routes, net_keys, omit_default_routes=True):
     return routing_tables
 
 
+def build_and_minimise_routing_tables(routes, net_keys, target_length=1023):
+    """Convert a set of RoutingTrees into a per-chip set of routing tables and
+    attempt to minimise those tables to meet a target length.
+
+    This is a wrapper around :py:func:`~.build_routing_tables` and
+    :py:func:`rig.routing_table.minimise`.
+
+    .. warning::
+        It may not be possible to minimise all tables such that they are
+        shorter than `target_length`. The chip co-ordinates of any routing
+        tables that could not be sufficiently minimised are reported as the
+        second returned value.
+
+    Parameters
+    ----------
+    routes : {net: :py:class:`~rig.place_and_route.routing_tree.RoutingTree`, \
+              ...}
+        The complete set of RoutingTrees representing all routes in the system.
+        (Note: this is the same datastructure produced by routers in the `par`
+        module.)
+    net_keys : {net: (key, mask), ...}
+        The key and mask associated with each net.
+    target_length : int
+        Target length of minimised routing tables.
+
+    Returns
+    -------
+    {(x, y): [:py:class:`~rig.routing_table.RoutingTableEntry`, ...]
+        Minimised routing tables. The correct operation of these tables relies
+        on their entries not being reordered.
+    {(x, y), ...}
+        Co-ordinates of chips for which tables shorter than `target_length`
+        could not be produced.
+    """
+    # Build the tables and then minimise them
+    tables = {
```

Comment:
Due to the CPU-bound nature of the problem and trivial parallelism available here it may be worth just dropping in: https://docs.python.org/2/library/multiprocessing.html#using-a-pool-of-workers as a quick and easy speedup.